sdk/eventhub/Azure.Messaging.EventHubs/src/EventProcessorClient.cs

```diff
@@ -522,146 +882,339 @@ protected override async ValueTask InitializeProcessingForPartitionAsync(Partiti
         }
 
         /// <summary>
-        ///   Updates the checkpoint using the given information for the associated partition and consumer group in the chosen storage service.
+        ///   Finds and tries to claim an ownership if this processor instance is eligible to increase its ownership list.
         /// </summary>
         ///
-        /// <param name="eventData">The event containing the information to be stored in the checkpoint.</param>
-        /// <param name="context">The context of the partition the checkpoint is associated with.</param>
+        /// <param name="partitionIds">The set of identifiers for the partitions within the Event Hub that this processor is associated with.</param>
+        /// <param name="completeOwnershipEnumerable">A complete enumerable of ownership obtained from the storage service provided by the user.</param>
+        /// <param name="activeOwnership">The set of ownership that are still active.</param>
         ///
-        /// <returns>A task to be resolved on when the operation has completed.</returns>
+        /// <returns>The claimed ownership. <c>null</c> if this instance is not eligible, if no claimable ownership was found or if the claim attempt failed.</returns>
         ///
-        internal Task InternalUpdateCheckpointAsync(EventData eventData,
-                                                    PartitionContext context) => UpdateCheckpointAsync(eventData, context);
+        private ValueTask<PartitionOwnership> FindAndClaimOwnershipAsync(string[] partitionIds,
```

Comment:
Will this ever run async?   I could be overlooking something but it appears to always be synchronous.   Since its a private call, if sync why would we want to return `ValueTask` instead of just the result?

Reply:
If we are not eligible to steal/claim ownership, this method will return synchronously (it happens most of the time because usually the processor is balanced). Otherwise, it will call `ClaimOwnership` in the `PartitionManager`, which is an `async` method. It has three possible `return` paths.

Reply:
Ah... gotcha.   I missed that path.   I see it now.   It was the direct rewrapping of the task into `ValueTask` that threw me.